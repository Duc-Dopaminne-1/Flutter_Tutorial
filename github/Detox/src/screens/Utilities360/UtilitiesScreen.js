import React from 'react';
import {FlatList, StyleSheet, View} from 'react-native';

import {IMAGES} from '../../assets/images';
import {translate} from '../../assets/localize';
import {COLORS} from '../../assets/theme/colors';
import BaseScreen from '../../components/BaseScreen';
import CustomListItem from '../../components/CustomListItem';
import ScreenIds from '../ScreenIds';

const getData = () => [
  {
    title: translate('utilities360.map360'),
    image: IMAGES.IC_MAP_FILL,
    screen: ScreenIds.Map360,
  },
  {
    title: translate('utilities360.valuatation'),
    image: IMAGES.IC_VALUATION_FILL,
    screen: ScreenIds.Valuatation,
  },
];

const keyExtractor = item => item.screen;

const UtilitiesScreen = ({navigation}) => {
  const DATA = getData();
  const onPressItem = screen => {
    if (screen) {
      navigation.navigate(screen);
    }
  };

  const renderItem = ({item}) => {
    return (
      <CustomListItem
        title={item.title}
        imageStyle={{tintColor: COLORS.PRIMARY_A100}}
        imageSource={item.image}
        onPress={() => onPressItem(item.screen)}
      />
    );
  };

  return (
    <BaseScreen testID={ScreenIds.Utilities360} title={translate('utilities360.headerTitle')}>
      <View style={styles.viewContainer}>
        <FlatList
          data={DATA}
          keyExtractor={keyExtractor}
          renderItem={renderItem}
          alwaysBounceVertical={false}
        />
      </View>
    </BaseScreen>
  );
};

const styles = StyleSheet.create({
  viewContainer: {
    marginBottom: 70,
  },
});

export default UtilitiesScreen;
