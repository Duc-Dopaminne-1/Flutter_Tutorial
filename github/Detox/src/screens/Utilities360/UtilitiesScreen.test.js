import React from 'react';
import renderer from 'react-test-renderer';

import UtilitiesScreen from './UtilitiesScreen';

jest.mock('../Auth/useLogin', () => {
  return {
    useLogin: () => ({
      showLogin: jest.fn(),
    }),
  };
});

it('renders correctly', () => {
  const tree = renderer.create(<UtilitiesScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
