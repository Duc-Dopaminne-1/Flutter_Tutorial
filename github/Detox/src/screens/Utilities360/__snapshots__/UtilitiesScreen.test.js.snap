// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#F9FAFB",
        "flex": 1,
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        undefined,
      ],
      Object {},
    ]
  }
  testID="UtilitiesScreen"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F9FAFB",
          "flex": 1,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          Object {
            "zIndex": 99,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 16,
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "zIndex": 99,
            }
          }
        >
          <View
            accessibilityLabel="BackButton"
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "paddingEnd": 12,
                "paddingVertical": 8,
                "zIndex": 99,
              }
            }
            testID="BackButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/images/arrow_left_linear.png",
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#313131",
                  "flex": 1,
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 24,
                  "includeFontPadding": true,
                },
                Object {},
                null,
              ]
            }
          >
            Tiện ích 360
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 70,
        }
      }
    >
      <RCTScrollView
        alwaysBounceVertical={false}
        data={
          Array [
            Object {
              "image": Object {
                "testUri": "../../../src/assets/images/ic_map_fill.png",
              },
              "screen": "Map360",
              "title": "Quy hoạch",
            },
            Object {
              "image": Object {
                "testUri": "../../../src/assets/images/ic_valuation_fill.png",
              },
              "screen": "Valuatation",
              "title": "Thẩm định giá",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  false,
                  Array [
                    Object {
                      "padding": 8,
                    },
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 8,
                        "marginHorizontal": 16,
                        "marginVertical": 8,
                        "paddingLeft": 12,
                        "paddingRight": 15,
                        "paddingVertical": 16,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "borderRadius": 20,
                        "justifyContent": "center",
                      },
                      Object {
                        "backgroundColor": "#D0E6CC",
                        "height": 32,
                        "width": 32,
                      },
                    ]
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={
                      Object {
                        "testUri": "../../../src/assets/images/ic_map_fill.png",
                      }
                    }
                    style={
                      Object {
                        "tintColor": "#26894F",
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "marginVertical": 6,
                        "paddingLeft": 8,
                      },
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "fontSize": 16,
                          },
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                          },
                          Object {
                            "fontFamily": "Nunito-Regular",
                            "fontSize": 15,
                            "includeFontPadding": true,
                          },
                        ],
                      ]
                    }
                  >
                    Quy hoạch
                  </Text>
                </View>
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/assets/images/arrow_right_linear.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "height": 20,
                        "width": 20,
                      },
                      Object {
                        "tintColor": "rgba(0, 0, 0, 0.54)",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  false,
                  Array [
                    Object {
                      "padding": 8,
                    },
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 8,
                        "marginHorizontal": 16,
                        "marginVertical": 8,
                        "paddingLeft": 12,
                        "paddingRight": 15,
                        "paddingVertical": 16,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "borderRadius": 20,
                        "justifyContent": "center",
                      },
                      Object {
                        "backgroundColor": "#D0E6CC",
                        "height": 32,
                        "width": 32,
                      },
                    ]
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={
                      Object {
                        "testUri": "../../../src/assets/images/ic_valuation_fill.png",
                      }
                    }
                    style={
                      Object {
                        "tintColor": "#26894F",
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "marginVertical": 6,
                        "paddingLeft": 8,
                      },
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "fontSize": 16,
                          },
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                          },
                          Object {
                            "fontFamily": "Nunito-Regular",
                            "fontSize": 15,
                            "includeFontPadding": true,
                          },
                        ],
                      ]
                    }
                  >
                    Thẩm định giá
                  </Text>
                </View>
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/assets/images/arrow_right_linear.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "height": 20,
                        "width": 20,
                      },
                      Object {
                        "tintColor": "rgba(0, 0, 0, 0.54)",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;
